llvm/lib/Target/SPIRV/SPIRVPrepareFunctions.cpp

```diff
@@ -314,39 +263,101 @@ static void substituteIntrinsicCalls(Module *M, Function *F) {
       auto *II = cast<IntrinsicInst>(Call);
       if (II->getIntrinsicID() == Intrinsic::memset ||
           II->getIntrinsicID() == Intrinsic::bswap)
-        lowerIntrinsicToFunction(M, II);
+        lowerIntrinsicToFunction(II);
       else if (II->getIntrinsicID() == Intrinsic::fshl ||
                II->getIntrinsicID() == Intrinsic::fshr)
-        lowerFunnelShifts(M, II);
+        lowerFunnelShifts(II);
       else if (II->getIntrinsicID() == Intrinsic::umul_with_overflow)
-        lowerUMulWithOverflow(M, II);
+        lowerUMulWithOverflow(II);
+      Changed = true;
```

Comment:
Should this `Changed = true;` be executed for all intrinsics (and not just for lowered ones)?

Reply:
@iliya-diyachkov I assume that `Call->setTailCall(false)` (above) counts as modifying input IR Module.

Reply:
@iliya-diyachkov I checked now by comparing Module hashes (`structuralHash(M)`) and it looks like my assumption was wrong. `setTailCall(false)` has no impact on module hash. Also `setTailCall(false)` sets an enum internally and that enum is already set by default. I will change this in a moment.

Reply:
@michalpaszkowski thanks. Also it's not clear to me why `Call->setTailCall(false);` is required here. I copied it from the translator, but now it looks redundant.